services/web/server/src/simcore_service_webserver/products_middlewares.py

```diff
@@ -1,24 +1,25 @@
 import logging
-from typing import Optional
 
 from aiohttp import web
 from servicelib.aiohttp.typing_extension import Handler
 
 from ._constants import APP_PRODUCTS_KEY, RQ_PRODUCT_KEY, X_PRODUCT_NAME_HEADER
 from ._meta import API_VTAG
+from .products import Product
 
 log = logging.getLogger(__name__)
 
 
-def discover_product_by_hostname(request: web.Request) -> Optional[str]:
-    for product in request.app[APP_PRODUCTS_KEY].values():
+def discover_product_by_hostname(request: web.Request) -> str | None:
+    products: dict[str, Product] = request.app[APP_PRODUCTS_KEY]
+    for _, product in products.items():
         if product.host_regex.search(request.host):
-            return product.name
+            return f"{product.name}"
```

Comment:
If `product.name` is already a string, this does the job. Not sure what is better though.
```suggestion
            product_name: str = product.name
            return product_name
```

Reply:
thanks! modified it and is working